lib/Jit/LLILCJit.cpp

```diff
@@ -274,15 +274,20 @@ CorJitResult LLILCJit::compileMethod(ICorJitInfo *JitInfo,
     }
     TargetOptions Options;
     CodeGenOpt::Level OptLevel;
-    if (Context.Options->OptLevel != ::OptLevel::DEBUG_CODE) {
+    bool IsNgen = Context.Flags & CORJIT_FLG_PREJIT;
+    bool IsReadyToRun = Context.Flags & CORJIT_FLG_READYTORUN;
+    if ((Context.Options->OptLevel != ::OptLevel::DEBUG_CODE) || IsNgen ||
```

Comment:
This feels like it will be a target specific thing. I'd say it at least deserves a comment saying why we are setting the opt levels like this for x64.